model/user-agent/registry.yaml

```diff
@@ -35,3 +35,20 @@ groups:
           using a user-agent for non-browser products, such as microservices with multiple names/versions inside the
           `user_agent.original`, the most significant version SHOULD be selected. In such a scenario it should align
           with `user_agent.name`
+      - id: user_agent.synthetic
+        stability: experimental
+        brief: >
+          A flag indicating that the user agent represents a synthetic source and did not originate from genuine client traffic.
+        note: >
+          This flag can primarily be determined by the contents of the `user_agent.original` attribute. Instrumentations should determine what they consider synthetic or bot traffic,
```

Comment:
It there any prior art we can refer to? E.g. a well-known database of crawlers/bots?

Reply:
There are a number sites that maintain lists of the most popular. So far, the best I've found is Data Dome's list of those most popular in 2024 https://datadome.co/bot-management-protection/crawlers-list/.

Reply:
cc @open-telemetry/javascript-approvers

Reply:
> I'd expect the attribute to be populated by a processor (in the SDK or collector) - this way it's easier to update and modify configuration (e.g. add new user-agents to the list of bots).

this is a good point, though I'm not sure if it would work for capturing `user_agent.synthetic.type` on metrics?

Reply:
oh, right. It's hard to imagine though that we'd add a static regex/dictionary to individual instrumentation libraries in all languages to populate something as dynamic as list of bots. E.g. this repo https://github.com/omrilotan/isbot updated patterns ~20 times during this year alone https://github.com/omrilotan/isbot/commits/main/src/patterns.json

Also it's not the only thing people would want to extract from the user-agent.
It seems to be a common thing to parse user_agent and build analytics on top of its components (e.g. https://github.com/open-telemetry/semantic-conventions/pull/1434 is about adding OS component). ECS defines a few other properties to extract  - https://www.elastic.co/guide/en/ecs/current/ecs-user_agent.html

We could try making the whole `user_agent.original` opt-in on metrics with expectation that processor would extract individual parts and remove the original string, but the local memory consumption could be too high due to cardinality. 

Another approach is that instrumentations might have a hook accepting external UserAgent parser that should be able to populate multiple attributes from the string.

Reply:
I like the greater scope of your proposed wording so we don't rule out any relevant components which might want to set this value.